tests/library/test_readme.py

```diff
@@ -0,0 +1,39 @@
+import re
+import unittest
+from typing import List
+
+from recipe_scrapers import SCRAPERS
+
+SEPARATOR = "Scrapers available for:"
+
+
+def get_documented_scrapers() -> List[str]:
+    documented_scrapers = []
+    with open("README.rst") as f:
+        found = False
+        for line in f:
+            if line.strip() == SEPARATOR:
+                found = True
+                next(f)  # Skip the underline
+            elif found and line.strip():
+                match = re.search(r"^- `http(?:s)?:\/\/(?:www.)?([^\s\/]+)", line)
+                if not match:
+                    break
+
+                domain = match.group(1)
+
+                documented_scrapers.append(domain)
+
```

Comment:
```suggestion
                continue
            if not found or not line.strip():
                continue

            match = re.search(r"^- `https?://(?:www\.)?([^/\s]+)", line)
            if match:
                documented_scrapers.append(match.group(1))
            else:
                # Stops reading if no longer within the scraper list section
                if found and line.startswith('(*) offline saved files only'):

```
some possible optimizations:

- slight changes to the regex, replaced (?:s)? with s?, changed escaping of slashes (\/) to simply /, changed ([^\s\/]+) to ([^/\s]+) to match the domain name without allowing forward slashes or blank spaces.
- simplified the conditional statements by replacing nested ifs with continue statements
- added hard stop to the logic on the off chance a matching url ends up lower in the readme causing issues.